src/PrestaShopBundle/Translation/TranslatorAwareTrait.php

```diff
@@ -0,0 +1,50 @@
+<?php
+/**
+ * 2007-2018 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShopBundle\Translation;
+
+use Symfony\Component\Translation\TranslatorInterface;
+
+/**
+ * Trait TranslatorAwareTrait is used for services that depends on translator
+ */
+trait TranslatorAwareTrait
```

Comment:
I'm not fan of traits, let me think about it... wdyt @eternoendless ?

Reply:
if you think it's not worth having trait for translator, it's fine for me. However, are there some drawbacks with traits?

Reply:
Let's keep it as it, for now, we have 3 or 4 months before 1.7.5 to see if it's a good or bad idea.

Personally, I prefer to inject required dependencies in contructor.